File: 0768/768.py
class Solution:
    def max_chunks_to_make_sorted_ii(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.max_chunks_to_make_sorted_ii(root.left), self.max_chunks_to_make_sorted_ii(root.right))

File: 0768/768.cpp
class Solution {
public:
    int maxChunksToMakeSortedIi(TreeNode* root) {
        if(!root) return 0;
        return 1+max(maxChunksToMakeSortedIi(root->left),maxChunksToMakeSortedIi(root->right));
    }
};